shell/platform/linux/public/flutter_linux/fl_application.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_PUBLIC_FLUTTER_LINUX_FL_APPLICATION_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_PUBLIC_FLUTTER_LINUX_FL_APPLICATION_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include "fl_plugin_registry.h"
+#include "fl_view.h"
+
+#include <gmodule.h>
+#include <gtk/gtk.h>
+
+G_BEGIN_DECLS
+
+G_MODULE_EXPORT
+G_DECLARE_DERIVABLE_TYPE(FlApplication,
+                         fl_application,
+                         FL,
+                         APPLICATION,
+                         GtkApplication);
+
+/**
+ * FlApplicationClass:
+ * @register_plugins: invoked when plugins should be registered.
+ */
+struct _FlApplicationClass {
+  GtkApplicationClass parent_class;
+
+  /**
+   * FlApplication::register_plugins:
+   * @application: the application
+   * @registry: registry to use.
+   *
+   * The ::register_plugins signal is emitted when plugins can be registered.
+   */
+  void (*register_plugins)(FlApplication* application,
+                           FlPluginRegistry* registry);
+};
+
+/**
+ * FlApplication:
+ *
+ * #Flutter-based application with the GTK embedder.
+ *
+ * Provides default behaviour for basic Flutter applications.
+ */
+
+/**
+ * fl_application_new:
+ * @application_id: (allow-none): The application ID or %NULL.
+ * @flags: The application flags.
+ *
+ * Creates a new Flutter-based application.
+ *
+ * Returns: a new #FlApplication
+ */
+FlApplication* fl_application_new(const gchar* application_id,
+                                  GApplicationFlags flags);
```

Comment:
Will we never need to add more parameters to `fl_application_new`?

From my understanding, if we need to add more parameters that would be a breaking change. We would need to migrate all Linux apps. If there's any risk of this needing more parameters in the future, perhaps this should accept a struct instead?

Reply:
I was also thinking about this. If we did need new parameters we have two options:
- Add some setters i.e. `fl_application_set_foo`.
- Make a new constructor i.e. `fl_application_new_with_foo`.
These are both common patterns in GObject code. The use of structs is not.

I think we're appropriately covered with the above. Anything new would likely fit well with a setter, as they are going to be optional and the application doesn't do anything until it is run so there's no requirement for it to be in the constructor.